Authentication events policy docs for user flow invocation (#10224)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/authenticationeventspolicy-put-onsignupstart.md

> ---

Reviewer: I think this should be update, not put.

> title: "Replace onSignupStart"

Reviewer: This should be Update, not Replace.

> description: "Replace an authenticationAction object."

Reviewer: The title and description should not appear to be saying different things.

author: "jkdouglas"
localization_priority: Normal
ms.prod: "microsoft-identity-platform"
doc_type: apiPageType
---

> # Replace onSignupStart

Reviewer: Update onSignupStart

Namespace: microsoft.graph

> Create a new authenticationAction object.

Reviewer 2: PUT here is not creating an object. It is changing an existing (identified by {id}) object.

Reviewer 3: PUT is typically "update the specified (authenticationAction) object is it already exists, create it otherwise as specified". Does this PUT action do the same?

Reviewer 3: BTW, if this operation is indeed "update specified object if exists, create it otherwise", then I think the filename should be   authenticationaction-put.md.

@jkdouglas , I don't suppose the stub topic generator generated this topic (PUT is not typical in the set of CRUD it generates), did it?

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from most to least privileged)|
|:---|:---|
|Delegated (work or school account)|Policy.ReadWrite.ApplicationConfiguration|
|Delegated (personal Microsoft account)|Not supported.|
|Application|Policy.ReadWrite.ApplicationConfiguration|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->

``` http
PUT /identity/events/onSignupStart/{id}
```

## Request headers

|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|
|Content-Type|application/json. Required.|

## Request body

In the request body, supply a JSON representation of the [authenticationAction](../resources/authenticationaction.md) object.

> The following table shows the properties that are required when you create the [invokeUserFlowAction](../resources/invokeuserflowaction.md).

Reviewer: I think this might also be another mapping problem for a user. The file name, title, and description refer to onSignupStart, authenticationAction, and invokeUserFlowAction. Is the user going to know the relationship of these when looking at this page?

|Property|Type|Description|
|:---|:---|:---|

> |priority|Int32|The priority of the action which is used for determining one out of multiple applicable actions.|

Reviewer: ```suggestion
|priority|Int32|The priority of the action that is used to determine one out of multiple applicable actions.|
```

> |sourceFilter|[authenticationSourceFilter](../resources/authenticationsourcefilter.md)|Filter based on the source of the authentication which is used to determine whether the action is executed or not.|

Reviewer: ```suggestion
|sourceFilter|[authenticationSourceFilter](../resources/authenticationsourcefilter.md)|Filter based on the source of the authentication that is used to determine whether the action is executed.|
```

> |userFlow|[b2xIdentityUserFlow](../resources/b2xidentityuserflow.md)|The reference to the user flow object that will be invoked in this action.|

Reviewer: ```suggestion
|userFlow|[b2xIdentityUserFlow](../resources/b2xidentityuserflow.md)|The reference to the user flow object that is invoked in this action.|
```

## Response

If successful, this method returns a `204 No Content` response code.

## Examples

### Request

<!-- {
  "blockType": "request",
  "name": "put_authenticationaction_from_"
}
-->

``` http

> POST https://graph.microsoft.com/beta/identity/events/onSignupStart

Reviewer 2: PUT instead of POST

Reviewer 3: Line 17:
Is that request query missing an {id} at the end, to specify the authenticationAction object?
